opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -354,12 +359,27 @@ namespace Opm
                     // q_gr = 1 / (b_g * d) * (q_gs - rs * q_os)
 
                     const double d = 1.0 - getValue(rv) * getValue(rs);
+
+                    if (d <= 0.0) {
+                    std::ostringstream sstr;
+                    sstr << "Problematic d value " << d << " obtained for well " << this->name()
+                         << " during computePerfRate calculations with rs " << rs
+                         << ", rv " << rv << " and pressure " << pressure
+                         << " obtaining d " << d
+                         << " Continue as if no dissolution (rs = 0) and vaporization (rv = 0) "
+                         << " for this connection.";
+                    deferred_logger.debug(sstr.str());
+                    }
                     // vaporized oil into gas
                     // rv * q_gr * b_g = rv * (q_gs - rs * q_os) / d
-                    perf_vap_oil_rate = getValue(rv) * (getValue(cq_s[gasCompIdx]) - getValue(rs) * getValue(cq_s[oilCompIdx])) / d;
+                    if (d > 0.0) {
```

Comment:
Same here, the two `if (d > 0.0)` can be in the `else` branch of `if (d <= 0.0) {`.